### PR TITLE
UWSHIEVMOD-83: stop including error file or incomplete package to pac…

### PR DESCRIPTION
**Pull request by Tao-Liu**
…kage creating

### CODE REVIEW

File: app/models/data_file.rb [hunk 1]
```diff
@@ -359,6 +359,14 @@ def is_error_file?
     self.file_processing_status.eql? STATUS_ERROR
   end
 
+  def is_incomplete_package?
+    if self.file_processing_status.eql? STATUS_PACKAGE
+      return [RESQUE_FAILED, RESQUE_WORKING, RESQUE_QUEUED].include? self.transfer_status
+    else
+      return nil
```
Comment (dyazbek-intersect) [inline app/models/data_file.rb]: Methods that end in '?' should only return true or false.

File: app/models/data_file.rb [hunk 2]
```diff
@@ -359,6 +359,14 @@ def is_error_file?
     self.file_processing_status.eql? STATUS_ERROR
   end
 
+  def is_incomplete_package?
```
Comment (dyazbek-intersect) [inline app/models/data_file.rb]: Shouldn't this method be in the Package class??

File: app/controllers/packages_controller.rb [hunk 3]
```diff
@@ -217,7 +217,9 @@ def validate_file_ids(file_ids, current_user, errors)
           errors << "file with id '#{file_id}' could not be found"
         else
           data_file = DataFile.find(id_as_int)
-          if data_file.is_authorised_for_access_by?(current_user)
+          if data_file.is_incomplete_package? || data_file.is_error_file?
+            errors << "unable to package error file '#{file_id}'"
```
Comment (dyazbek-intersect) [inline app/controllers/packages_controller.rb]: Bad grammar in the error message; it should be something like ''file '#{file_id}' is not in a state that can be packaged"

File: app/controllers/packages_controller.rb [hunk 4]
```diff
@@ -217,7 +217,9 @@ def validate_file_ids(file_ids, current_user, errors)
           errors << "file with id '#{file_id}' could not be found"
         else
           data_file = DataFile.find(id_as_int)
-          if data_file.is_authorised_for_access_by?(current_user)
+          if data_file.is_incomplete_package? || data_file.is_error_file?
```
Comment (dyazbek-intersect) [inline app/controllers/packages_controller.rb]: I think you should check if there is a package of the same ID and only do this check if it is infact a package.

File: features/api_package_create.feature [hunk 5]
```diff
@@ -14,6 +14,10 @@ Feature: Create a package from the API
       | sample1.txt | 01/12/2011 13:45 | sean@intersect.org.au  | 1/6/2010 6:42:01 | 30/11/2011 18:05:23 | samples/sample1.txt | 1  |
       | sample2.txt | 30/11/2011 10:15 | admin@intersect.org.au | 1/6/2010 6:42:01 | 30/11/2011 18:05:23 | samples/sample2.txt | 2  |
       | sample3.txt | 30/12/2011 12:34 | admin@intersect.org.au |                  |                     | samples/sample3.txt | 3  |
+      | error_file.txt | 30/12/2011 12:34 | admin@intersect.org.au |                  |                     | samples/error_file.txt | 4  |
+      | incomplete_package.zip | 30/12/2011 12:34 | admin@intersect.org.au |                  |                     | samples/incomplete_package.zip | 5  |
+    And I set file "error_file.txt" as error file
```
Comment (dyazbek-intersect) [inline features/api_package_create.feature]: You should change this method to make it more generic... Something like:
And I set the file processing status to "ERROR" for file "error_file.txt"
This way, we can use the same function to set any file processing status....

File: features/api_package_create.feature [hunk 6]
```diff
@@ -14,6 +14,10 @@ Feature: Create a package from the API
...
+    And I set package "incomplete_package.zip" as "FAILED"
```
Comment (dyazbek-intersect) [inline features/api_package_create.feature]: Same as above comment.

**Comment by dyazbek-intersect** [comment]
Please address comments.

**Comment by Tao-Liu** [comment]
Comments addressed, please review again

**Comment by dyazbek-intersect** [comment]
Looks good. Be sure to pull before you commit - there will be some conflicts....

**Pull request merged**